package.json

```diff
@@ -1,23 +1,61 @@
 {
   "name": "skyward",
   "version": "1.0.0",
-  "main": "node_modules/expo/AppEntry.js",
+  "main": "expo-router/entry",
   "scripts": {
     "start": "expo start",
     "android": "expo start --android",
     "ios": "expo start --ios",
-    "web": "expo start --web"
+    "web": "expo start --web",
+    "test": "jest --watch --coverage=false --changedSince=origin/main",
+    "testDebug": "jest -o --watch --coverage=false",
+    "testFinal": "jest"
   },
   "dependencies": {
+    "@reduxjs/toolkit": "^2.1.0",
+    "@shopify/flash-list": "1.6.3",
     "expo": "~50.0.6",
+    "expo-constants": "~15.4.5",
+    "expo-image": "~1.10.6",
+    "expo-linear-gradient": "~12.7.1",
+    "expo-linking": "~6.2.2",
+    "expo-router": "~3.4.7",
     "expo-status-bar": "~1.11.1",
+    "jest": "^29.3.1",
+    "jest-expo": "~50.0.2",
+    "moment": "^2.30.1",
     "react": "18.2.0",
-    "react-native": "0.73.4"
+    "react-native": "0.73.4",
+    "react-native-safe-area-context": "4.8.2",
+    "react-native-screens": "~3.29.0",
+    "react-redux": "^9.1.0",
+    "redux": "^5.0.1"
   },
   "devDependencies": {
     "@babel/core": "^7.20.0",
+    "@testing-library/react-native": "^12.4.3",
+    "@types/jest": "^29.5.12",
     "@types/react": "~18.2.45",
+    "msw": "^2.2.0",
     "typescript": "^5.1.3"
   },
-  "private": true
+  "private": true,
+  "jest": {
+    "preset": "jest-expo",
+    "transformIgnorePatterns": [
+      "node_modules/(?!((jest-)?react-native|@react-native(-community)?)|expo(nent)?|@expo(nent)?/.*|@expo-google-fonts/.*|react-navigation|@react-navigation/.*|@unimodules/.*|unimodules|sentry-expo|native-base|react-native-svg|react-redux)"
+    ],
+    "collectCoverage": true,
+    "collectCoverageFrom": [
+      "**/*.{ts,tsx}",
+      "!**/coverage/**",
+      "!**/node_modules/**",
+      "!**/babel.config.js",
+      "!**/jest.setup.js",
+      "!**/_layout.{ts,tsx}"
```

Comment:
I removed the `_layout` files from coverage since I haven't found a reliable way to test it using expo-router, in my perspective the navigation can be tested in a more reliable way using e2e tests tools like detox.